tests/src/test/scala/cats/tests/AlgebraInvariantTests.scala

```diff
@@ -22,21 +23,159 @@ class AlgebraInvariantTests extends CatsSuite {
   val genMonoidInt: Gen[Monoid[Int]] =
     Gen.oneOf(implicitly[Monoid[Int]], intMultiplication, maxInt)
 
+  val genCommutativeMonoidInt: Gen[CommutativeMonoid[Int]] =
+    Gen.oneOf(implicitly[CommutativeMonoid[Int]], intMultiplication)
+
+  val genBoundedSemilatticeSetInt: Gen[BoundedSemilattice[Set[Int]]] =
+    Gen.const(implicitly[BoundedSemilattice[Set[Int]]])
+
+  val genCommutativeGroupInt: Gen[CommutativeGroup[Int]] =
+    Gen.const(implicitly[CommutativeGroup[Int]])
+
   implicit val arbMonoidInt: Arbitrary[Monoid[Int]] =
     Arbitrary(genMonoidInt)
 
-  implicit val arbSemigoupInt: Arbitrary[Semigroup[Int]] =
+  implicit val arbSemigroupInt: Arbitrary[Semigroup[Int]] =
     Arbitrary(genMonoidInt)
 
+  implicit val arbCommutativeMonoidInt: Arbitrary[CommutativeMonoid[Int]] =
+    Arbitrary(genCommutativeMonoidInt)
+
+  implicit val arbCommutativeSemigroupInt: Arbitrary[CommutativeSemigroup[Int]] =
+    Arbitrary(genCommutativeMonoidInt)
+
+  implicit val arbBandSetInt: Arbitrary[Band[Set[Int]]] =
+    Arbitrary(genBoundedSemilatticeSetInt)
+
+  implicit val arbSemilatticeSetInt: Arbitrary[Semilattice[Set[Int]]] =
+    Arbitrary(genBoundedSemilatticeSetInt)
+
+  implicit val arbBoundedSemilatticeSetInt: Arbitrary[BoundedSemilattice[Set[Int]]] =
+    Arbitrary(genBoundedSemilatticeSetInt)
+
+  implicit val arbGroupInt: Arbitrary[Group[Int]] =
+    Arbitrary(genCommutativeGroupInt)
+
+  implicit val arbCommutativeGroupInt: Arbitrary[CommutativeGroup[Int]] =
+    Arbitrary(genCommutativeGroupInt)
+
+  def combineAllOption_Ref[F[_] : InvariantMonoidal, A: Eq](name: String, a: A)(implicit ev: F[A] <:< Semigroup[A]): Unit = {
```

Comment:
Hmm, I was suggesting adding this to the `SemigroupLaw` but I just realized that the consolidation PR #1922 isn't merged yet. Let's wait until that one gets merged, shall we?

Reply:
I agree we should wait, however, this one isn't really about `combineAllOption` for any given semigroup, but for the ones created by `InvariantMonoidal[F].pure`. `SemigroupLaws` already has a check for `combineAllOption` here: https://github.com/typelevel/cats/blob/master/kernel-laws/src/main/scala/cats/kernel/laws/GroupLaws.scala#L33 :)

Reply:
ah I see.  so, even without this methods and the tests that use it, the `combineAllOption` should be covered once you used `GroupLaws` to test the new instances? So for these `pure` given instance, why not test them with the `GroupLaws` again?

Reply:
I think we could do that, but I didn't think the `pure` ones were actually lawful instances.

Reply:
I'd be curious to find out.

Reply:
Okay so the `Semigroup` instance is not lawful, because of the `combineAllOption`, as defined  [here](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/instances/semigroup.scala#L20), is not consistent with `reduceOption(_ |+|
 _)`, because when you pass a collection with only one value `reduceOption` will only turn that into an `Option`. One could argue that this doesn't necessarily break the Semigroup laws though.

In addition the `Monoid` instance fails to fulfil the left and right identity laws. 
Also these instances aren't able to abide by the idempotency law. 

Does this mean `Monoid` is not a valid `InvariantMonoidal`? I'm not sure I know enough about that.

Reply:
if the pure instance is not lawful, I am not sure if it's still a `InvariantMonoidal`, maybe just a `Invariant`? maybe worthwhile to find out how it's defined in Haskell.

Reply:
I was surprised at the `pure`. I thought these two were only `Invariant`.

By the way, I think `Invariant` and `Cartesian` are the two ways you actually want to compose these things.

Reply:
So, I think we should get rid of all Invariant Monoidal except for Semigroup and Commutative Semigroup and Turn the others into Invariant instances

Reply:
👍  BTW, for invariance instances, they can be moved the companion object of `Invariant` right?